### PR TITLE
writing basic form for details and approve form and connect it with the backend to do all needed tasks #53

### PR DESCRIPTION
**Pull request by Hadeel-Salamin**
- [x] write basic form to show details.

- [x] get the event id by passing it from calendar event props and saving it into DetailsEvent Component state.

- [x] send **GET** request to the server to **/event/:id** to get all the details that related to the event id was passed.

- [x] save the "data.data" you got from this endpoint into variables to use them later into the form for showing details.

- [x] make sure to make any date looks into _mm-dd-yyyy_ format.

- [x] use the status as flag to make this form for details once the event is approved and for approve if has not been approved yet.

- [x] to approve the event send **PUT** request to **/event/:id** to update the event status from 0 to 1.

### CODE REVIEW

File: client/src/components/Calendar/index.js [hunk 1]
```diff
@@ -41,17 +41,17 @@ class BigCalendar extends Component {
       });
   }
 
-  click = (event) => {
+  click = event => {
```
Comment (m4v15) [inline client/src/components/Calendar/index.js]: I don't think you have to do it here, but just a heads up that maybe these functions could have better names?

`click` => `bookNewEvent`
`show` => `seeEventDetails`

Also, out of interest, what does the second argument `event` do in `history.push`?

Comment (m4v15) [reply client/src/components/Calendar/index.js]: I guess it passes props to the pathname component or something? 

Also, I guess it comes from React Big Calendar, sah? When you click on something it passes the arguments - should it definitely be called `event` in this case, and not something like `dates`? It makes sense if it's in the `onSelectEvent` for it to be `event` but this function is assigned to `onSelectSlot`

File: client/src/components/DetailsEvent/index.js [hunk 2]
```diff
@@ -1,13 +1,180 @@
...
+          <div>
+            <NavBar />
+          </div>
+          <div className="both">
+            <SideBar />
+            <div className="detailsEvent">
+              <form onSubmit={this.handleSubmitForm}>
+                <h2 className="h2">General</h2>
+                <div className="labels_container">
+                  <label className="title">Event title:</label>
```
Comment (m4v15) [inline client/src/components/DetailsEvent/index.js]: Labels with no inputs? Unless you can edit these Maybe both should just be `div`s? Or am I missing something

File: client/src/components/DetailsEvent/index.js [hunk 3]
```diff
@@ -1,13 +1,180 @@
...
+                  <label className="answer">{coffee_price}</label>
+                </div>
+                <h2 className="h2">Equipment</h2>
+                <div className="labels_container">
+                  <label className="title">Description:</label>
+                  <label className="answer">{equipment_note}</label>
+                </div>
+                <div className="labels_container">
+                  <label className="title">price:</label>
+                  <label className="answer">{equipment_price}</label>
```
Comment (m4v15) [inline client/src/components/DetailsEvent/index.js]: This is all _very_ repetitive, can you find a way to loop over an array to do this?

File: client/src/components/DetailsEvent/index.js [hunk 4]
```diff
@@ -1,13 +1,180 @@
...
+    status: this.props.history.location.event.status,
+    loading: false
+  };
+
+  componentDidMount() {
+    axios
+      .get(`/event/${this.state.id}`)
+      .then(data => {
+        this.setState({
+          event_details: data.data.event,
```
Comment (m4v15) [inline client/src/components/DetailsEvent/index.js]: seems like maybe call it `apiResponse` on line 16 or deconstruct: `.then({ data } => {` so then you just have to do `data.event` here?

File: client/src/components/DetailsEvent/index.js [hunk 5]
```diff
@@ -1,13 +1,180 @@
...
+                <h2 className="h2">Equipment</h2>
+                <div className="labels_container">
+                  <label className="title">Description:</label>
+                  <label className="answer">{equipment_note}</label>
+                </div>
+                <div className="labels_container">
+                  <label className="title">price:</label>
+                  <label className="answer">{equipment_price}</label>
+                </div>
+                {this.state.status && <input type="submit" value="OK" />}
```
Comment (m4v15) [inline client/src/components/DetailsEvent/index.js]: Could pull all of this out into a function above the `render` function, which takes status, something like (syntax might be wrong):

```js
renderButtons = status => { 
  const button = status ?
   <input type="submit" value="OK" />
   : <input type="submit" value="Approve" />

return (
  { button}
 <input type="button" value="Cancel" onClick={this.cancel} />
)
}
```

Then in the render you just call this function and pass status, ```this.renderButtons(this.state.status)```

Comment (m4v15) [reply client/src/components/DetailsEvent/index.js]: Also I'm not sure about what happens here, do you only show the cancel button if the status doesn't exist? Maybe this is just me being confused, what does `status === true` actually mean? That it has been Approved?

File: client/src/components/DetailsEvent/index.js [hunk 6]
```diff
@@ -1,13 +1,180 @@
...
+          loading: true
+        });
+      })
+      .catch(() => {
+        const { history } = this.props;
+        history.push("/error");
+      });
+  }
+
+  handleSubmitForm = event => {
```
Comment (m4v15) [inline client/src/components/DetailsEvent/index.js]: Why not just make multiple submit functions, and pass the correct one depending on the status check you are doing down on line 166?

Like, `handleApprove`, `handleCancel` or something?

File: client/src/components/DetailsEvent/index.js [hunk 7]
```diff
@@ -1,13 +1,180 @@
...
-import Header from '../Header/header';
-class DetailsEvent extends React.Component {
+import React, { Component } from "react";
+import NavBar from "../NavBar";
+import SideBar from "../SideBar";
+import axios from "axios";
+
+class DetailsEvent extends Component {
+  state = {
+    id: this.props.history.location.event.id,
```
Comment (shahenazmonia) [inline client/src/components/DetailsEvent/index.js]: ```id``` is not a general variable to put it in the state, you have just to get the passed id from ```params```, so define it where you are going to use

File: client/src/components/DetailsEvent/index.js (hunk 2 above)
Comment (Hadeel-Salamin) [reply client/src/components/DetailsEvent/index.js]: yeah we have to change this, will be div and p I guess

File: client/src/components/DetailsEvent/index.js (hunk 5 above)
Comment (Hadeel-Salamin) [reply client/src/components/DetailsEvent/index.js]: yeah if status === 1 "true" that means this event has been approved so I will not show this button anymore, and will show the button with "ok" as it form to show details only.

File: client/src/components/Calendar/index.js (hunk 1 above)
Comment (m4v15) [reply client/src/components/Calendar/index.js]: Sorry to clarify this second comment, I'm asking what is the argument that this `click` function receives, which you call `event` here?

Is it the normal information about the clickEvent itself like [this](https://developer.mozilla.org/en-US/docs/Web/Events/click)? Or is it special info that ReactBigCalender gives to these actions

Comment (m4v15) [reply client/src/components/Calendar/index.js]: If we `console.log(event)` what does it look like

File: client/src/components/DetailsEvent/index.js (hunk 7 above)
Comment (Hadeel-Salamin) [reply client/src/components/DetailsEvent/index.js]: done

File: client/src/components/Calendar/index.js (hunk 1 above)
Comment (Hadeel-Salamin) [reply client/src/components/Calendar/index.js]: the event details we got from the GET /events endpoint
thats what we got by console.log("event",event):
event {id: 10, title: "kkkkk", start: Mon Feb 18 2019 09:49:20 GMT+0200 (Israel Standard Time), end: Tue Feb 19 2019 09:49:20 GMT+0200 (Israel Standard Time), status: 1, …}

Comment (Hadeel-Salamin) [reply client/src/components/Calendar/index.js]: "`click` => `bookNewEvent`
`show` => `seeEventDetails` "
changed

Comment (m4v15) [reply client/src/components/Calendar/index.js]: > the event details we got from the GET /events endpoint
> thats what we got by console.log("event",event):
> `event {id: 10, title: "kkkkk", start: Mon Feb 18 2019 09:49:20 GMT+0200 (Israel Standard Time), end: Tue Feb 19 2019 09:49:20 GMT+0200 (Israel Standard Time), status: 1, …}`

Is this when you console log it inside the `bookNewEvent` function or `seeEventDetails` - would it not be different for each?

Comment (m4v15) [reply client/src/components/Calendar/index.js]: Sorry, I don't want to hold this PR up - if you want to merge it, go ahead, we can talk about this tomorrow

**Pull request merged**